core/services/ocr2/plugins/ccip/estimatorconfig/interceptors/mantle/interceptor.go

```
@@ -0,0 +1,81 @@
+package mantle
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+
+	evmClient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/gas/rollups"
+)
+
+const (
+	// tokenRatio is not volatile and can be requested not often.
+	tokenRatioUpdateInterval = 60 * time.Minute
+	// tokenRatio fetches the tokenRatio used for Mantle's gas price calculation
+	tokenRatioMethod          = "tokenRatio"
+	mantleTokenRatioAbiString = `[{"inputs":[],"name":"tokenRatio","outputs":[{"internalType":"uint256","name":"","type":"uint256"}],"stateMutability":"view","type":"function"}]`
+)
+
+type Interceptor struct {
+	client               evmClient.Client
+	tokenRatioCallData   []byte
+	tokenRatio           *big.Int
+	tokenRatioLastUpdate time.Time
+}
+
+func NewInterceptor(_ context.Context, client evmClient.Client) *Interceptor {
+	// Encode calldata for tokenRatio method
+	tokenRatioMethodAbi, err := abi.JSON(strings.NewReader(mantleTokenRatioAbiString))
+	if err != nil {
+		log.Panicf("failed to parse GasPriceOracle %s() method ABI for Mantle; %v", tokenRatioMethod, err)
```

Why `log.Panicf`? If you want to log it then use logger. However, I think you should return an error from the contructor. Right now you will silently log errors and return broken struct, right?

---

It shouldn't return an error, @matYang was against it. But it not silently logging. It panic.

---

I don't think panic is a good approach, misconfig will bring the plugin down, what's wrong with returning error? This is exactly what we do in the parent function when something is misconfigured or fail to init

---

I think my suggestion was against making an RPC call and return an error in `NewInterceptor`, we don't want temp RPC issue to block plugin initialization, it does make sense to error if there is a misconfig

---

We don't call RPC in the constructor so we should return error instead of panic ;)

---

fixed